pysteps/blending/steps.py

```diff
@@ -65,1723 +64,2862 @@
+    std_models_timestep: Optional[np.ndarray] = None
+    velocity_models_timestep: Optional[np.ndarray] = None
+
+    # Mapping from NWP members to ensemble members
+    mapping_list_NWP_member_to_ensemble_member: Optional[np.ndarray] = None
+
+    # Random states for precipitation and motion
+    randgen_precip: Optional[List[np.random.RandomState]] = None
+    randgen_motion: Optional[List[np.random.RandomState]] = None
+
+    # Variables for final forecast computation
+    previous_displacement: Optional[List[Any]] = None
+    previous_displacement_noise_cascade: Optional[List[Any]] = None
+    previous_displacement_prob_matching: Optional[List[Any]] = None
+    rho_final_blended_forecast: Optional[np.ndarray] = None
+    final_blended_forecast_means: Optional[np.ndarray] = None
+    final_blended_forecast_stds: Optional[np.ndarray] = None
+    final_blended_forecast_means_mod_only: Optional[np.ndarray] = None
+    final_blended_forecast_stds_mod_only: Optional[np.ndarray] = None
+    final_blended_forecast_cascades: Optional[np.ndarray] = None
+    final_blended_forecast_cascades_mod_only: Optional[np.ndarray] = None
+    final_blended_forecast_recomposed: Optional[np.ndarray] = None
+    final_blended_forecast_recomposed_mod_only: Optional[np.ndarray] = None
+
+    # Final outputs
+    final_blended_forecast: Optional[np.ndarray] = None
+    final_blended_forecast_non_perturbed: Optional[np.ndarray] = None
+
+    # Timing and indexing
+    time_prev_timestep: Optional[List[float]] = None
+    leadtime_since_start_forecast: Optional[List[float]] = None
+    subtimesteps: Optional[List[float]] = None
+    is_nowcast_time_step: Optional[bool] = None
+    subtimestep_index: Optional[int] = None
+
+    # Weights used for blending
+    weights: Optional[np.ndarray] = None
+    weights_model_only: Optional[np.ndarray] = None
+
+    # Is stores here aswell becasue this is changed during the forecast loop and thus not part of the config
+    extrapolation_kwargs: Dict[str, Any] = field(default_factory=dict)
+
+
+class StepsBlendingNowcaster:
+    def __init__(
+        self,
+        precip,
+        precip_models,
+        velocity,
+        velocity_models,
+        time_steps,
+        issue_time,
+        steps_blending_config: StepsBlendingConfig,
+    ):
+        """Initializes the StepsBlendingNowcaster with inputs and configurations."""
+        # Store inputs
+        self.__precip = precip
+        self.__precip_models = precip_models
+        self.__velocity = velocity
+        self.__velocity_models = velocity_models
+        self.__timesteps = time_steps
+        self.__issuetime = issue_time
+
+        self.__config = steps_blending_config
+
+        # Initialize Params and State
+        self.__params = StepsBlendingParams()
+        self.__state = StepsBlendingState()
+
+        # Additional variables for time measurement
+        self.__start_time_init = None
+        self.__zero_precip_time = None
```

Comment:
This variable appears to be unused

Reply:
Thanks, I should indeed use this one, currently that is a local variable :) I changed this in the new version